app/main.py

```diff
@@ -1,6 +1,17 @@
 from typing import Callable
 
 
-def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+def cache(*func: Callable) -> Callable:
+    result = {}
+    for fun in func:
+        def decorate(*args) -> None:
+            if len(result) == 0 or args not in result:
+                print("Calculating new result")
+                new_value = fun(*args)
+                result[args] = new_value
+                return new_value
+            else:
+                print("Getting from cache")
+                return result[args]
+            return result
+        return decorate
```

Comment:
The `cache` function is expected to return a `Callable`, but it currently returns only the last decorated function. Consider returning a list of decorated functions or modifying the logic to handle multiple functions correctly.